Library/Homebrew/rubocops/service.rb

```diff
@@ -21,15 +21,39 @@ class Service < FormulaCop
           share:    :opt_share,
         }.freeze
 
+        # These can be defined without using the `run` command.
+        RENAME_METHOD_CALLS = [:plist_name, :service_name].freeze
+
+        # At least one of these methods must be defined in a service block.
+        REQUIRED_METHOD_CALLS = [:run, *RENAME_METHOD_CALLS].freeze
+
         def audit_formula(_node, _class_node, _parent_class_node, body_node)
           service_node = find_block(body_node, :service)
           return if service_node.blank?
 
+          method_calls = service_node.each_descendant(:send).group_by(&:method_name)
+          method_calls.delete(:service)
+
+          if (method_calls.keys & REQUIRED_METHOD_CALLS).empty?
+            offending_node(service_node)
+            problem "Service blocks require `run`, `plist_name` or `service_name` to be defined."
+          end
+
+          unless method_calls.key?(:run)
+            other_method_calls = method_calls.keys - RENAME_METHOD_CALLS
+            if other_method_calls.any?
+              offending_node(service_node)
+              problem "`run` must be defined to use methods other than " \
+                      "`service_name` and `plist_name` like #{other_method_calls}."
+            end
+          end
```

Comment:
```suggestion
          if (method_calls.keys & REQUIRED_METHOD_CALLS).empty?
            offending_node(service_node)
            problem "Service blocks require `run`, `plist_name` or `service_name` to be defined."
          elsif !method_calls.key?(:run)
            other_method_calls = method_calls.keys - RENAME_METHOD_CALLS
            if other_method_calls.any?
              offending_node(service_node)
              problem "`run` must be defined to use methods other than " \
                      "`service_name` and `plist_name` like #{other_method_calls}."
            end
          end
```

On second thought, it probably doesn't make sense to show both of these linter errors at the same time because fixing the first problem might fix the second as well.

Reply:
Fixed on rebase.